src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConfigWidgets/SymbolsDialog.h"
+
+#include <qpushbutton.h>
+
+#include <QFileDialog>
+#include <QPushButton>
+#include <algorithm>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_SymbolsDialog.h"
+
+namespace orbit_config_widgets {
+
+SymbolsDialog::~SymbolsDialog() noexcept = default;
+
+SymbolsDialog::SymbolsDialog(QWidget* parent)
+    : QDialog(parent), ui_(std::make_unique<Ui::SymbolsDialog>()) {
+  ui_->setupUi(this);
+
+  QObject::connect(ui_->addButton, &QPushButton::clicked, this, [this]() {
+    QString directory = QFileDialog::getExistingDirectory(this, "Select Symbol Directory");
+    if (!directory.isEmpty()) {
+      AddSymbolPath(std::filesystem::path{directory.toStdString()});
+    }
+  });
+
+  QObject::connect(ui_->removeButton, &QPushButton::clicked, this,
+                   &SymbolsDialog::OnRemoveButtonClicked);
+
+  QObject::connect(ui_->listWidget, &QListWidget::itemSelectionChanged, this,
+                   &SymbolsDialog::OnSelectionChanged);
+
+  // Initially disabled, will be enabled once selection changes
+  ui_->removeButton->setEnabled(false);
+}
+
+void SymbolsDialog::SetSymbolPaths(const std::vector<std::filesystem::path>& paths) {
+  ui_->listWidget->clear();
+  QStringList paths_list;
+  paths_list.reserve(static_cast<int>(paths.size()));
+
+  std::transform(
+      paths.begin(), paths.end(), std::back_inserter(paths_list),
+      [](const std::filesystem::path& path) { return QString::fromStdString(path.string()); });
+
+  ui_->listWidget->addItems(paths_list);
+}
+
+void SymbolsDialog::AddSymbolPath(const std::filesystem::path& path) {
+  ui_->listWidget->addItem(QString::fromStdString(path.string()));
+}
+
+[[nodiscard]] std::vector<std::filesystem::path> SymbolsDialog::GetSymbolPaths() {
+  std::vector<std::filesystem::path> result;
+  result.reserve(ui_->listWidget->count());
+
+  for (int i = 0; i < ui_->listWidget->count(); ++i) {
+    result.emplace_back(ui_->listWidget->item(i)->text().toStdString());
+  }
+
+  return result;
+}
+
+void SymbolsDialog::OnRemoveButtonClicked() {
+  QList<QListWidgetItem*> selected_items = ui_->listWidget->selectedItems();
+
+  for (QListWidgetItem* selected_item : selected_items) {
+    ui_->listWidget->takeItem(ui_->listWidget->row(selected_item));
+  }
+}
+
+void SymbolsDialog::OnSelectionChanged() {
+  ui_->removeButton->setEnabled(false);
+  QList<QListWidgetItem*> selected_items = ui_->listWidget->selectedItems();
+
+  if (!selected_items.isEmpty()) {
+    ui_->removeButton->setEnabled(true);
+  }
```

Comment:
Isn't this just a one-liner `ui_->removeButton->setEnabled(!selected_items.isEmpty());`?

Reply:
Unless that is we do not want to disable the button, which I don't think the case here.

Reply:
done. Also inlined into the connect